src/client/renderer/r_particle.c

```diff
@@ -247,16 +350,25 @@ void R_UpdateParticles(r_element_t *e, const size_t count) {
 
 	for (i = 0; i < count; i++, e++) {
 
-		if (e->type == ELEMENT_PARTICLE) {
-			r_particle_t *p = (r_particle_t *) e->element;
+		if (e->type != ELEMENT_PARTICLE) {
+			continue;
+		}
+		
+		r_particle_t *p = (r_particle_t *) e->element;
+		
+		if (r_state.particle_program == program_particle) {
 
-			const uint16_t vertex_start = r_particle_state.num_particles * 4;
+			R_ParticleGeometryVerts(p, &r_particle_state.geometry_verts[r_particle_state.num_particles]);
+			R_ParticleGeometryTexcoords(p, &r_particle_state.geometry_verts[r_particle_state.num_particles]);
+			R_ParticleGeometryColor(p, &r_particle_state.geometry_verts[r_particle_state.num_particles]);
+		} else {
+			const uint32_t vertex_start = r_particle_state.num_particles * 4;
 
 			R_ParticleVerts(p, &r_particle_state.verts[vertex_start]);
 			R_ParticleTexcoords(p, &r_particle_state.verts[vertex_start]);
 			R_ParticleColor(p, &r_particle_state.verts[vertex_start]);
 
-			const uint16_t index_start = r_particle_state.num_particles * 6;
+			const uint32_t index_start = r_particle_state.num_particles * 6;
```

Comment:
Yea, I think back when we were using `GL_QUADS`, we used to be okay with `uint16_t` -- or at least we blew it out much less often ;)